File: app/src/main/java/com/ntl/guidelinesapp/modules/room_db/database/UserDAO.java
package com.ntl.guidelinesapp.modules.room_db.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.ntl.guidelinesapp.modules.room_db.User;

import java.util.List;

@Dao
public interface UserDAO {

    @Insert
    void insertUser(User user);

    @Query("SELECT * FROM user")
    List<User> getListUser();

    @Query("SELECT * FROM user where username= :username")
    List<User> checkUser(String username);

    @Update
    void updateUser(User user);

    @Delete
    void deleteUser(User user);

    @Query("DELETE FROM user")
    void dellAllUser();

    @Query("select * from user where username like '%' || :username || '%'")
    List<User> searchUsername(String username);
}

File: app/src/main/java/com/ntl/guidelinesapp/modules/room_db/database/UserDAO.kt
package com.ntl.guidelinesapp.modules.room_db.database

import androidx.room.*
import com.ntl.guidelinesapp.modules.room_db.User

@Dao
interface UserDAO {
    @Insert
    fun insertUser(user: User?)

    //coroutine
    @Insert
    suspend fun addUser(user: User?)

    @Query("SELECT * FROM user LIMIT 20")
    suspend fun getLast20User(): List<User>

    @get:Query("SELECT * FROM user")
    val listUser: List<User?>?

    @Query("SELECT * FROM user where username= :username")
    fun checkUser(username: String?): List<User?>?

    @Update
    fun updateUser(user: User?)

    @Delete
    fun deleteUser(user: User?)

    @Query("DELETE FROM user")
    fun dellAllUser()

    @Query("select * from user where username like '%' || :username || '%'")
    fun searchUsername(username: String?): List<User?>?
}